### DISCUSSION TITLE
How to run in TF12 ?

### DISCUSSION THREAD

**Discussion by zkarampa** [cyrilgdn/terraform-provider-postgresql#87] [Q&A] [answered]
Hello, 

The README mentions that requirements are: Terraform 0.12.x

However I'm not able to use the provider in TF12 (0.12.29) as TF12 doesn't support community providers.
Does anyone know any way or example on how to use it with TF12? 

Thanks!

**Comment by cyrilgdn** [comment, marked as answer]
@zkarampa The README is probably a bit outdated indeed, I'll update it. 

If you want to have recent versions, you should probably upgrade to Terraform 0.13.x or above.

But I guess it's not that simple in your case if you haven't done it yet.

So to answer your question, to use the `postgresql` provider (or other community providers) for Terraform 0.12.x, you have 2 solutions:
* Use the [last released version](https://github.com/hashicorp/terraform-provider-postgresql/releases/tag/v1.7.1) for Terraform 0.12.x 
* "Manually" download the binary of the latest version into one of the plugins directory.
   For example with a script like:

```bash
PLUGINS_DIR="${HOME}/.terraform.d/plugins"
VERSION=1.13.0

curl -Lo "/tmp/terraform-provider-postgresql_v${VERSION}.zip" \
    https://github.com/cyrilgdn/terraform-provider-postgresql/releases/download/v${VERSION}/terraform-provider-postgresql_${VERSION}_linux_amd64.zip
unzip -d "$PLUGINS_DIR" /tmp/terraform-provider-postgresql_v${VERSION}.zip terraform-provider-postgresql_v${VERSION}
chmod +x "${PLUGINS_DIR}/terraform-provider-postgresql_v${VERSION}"
```

**Reply by zkarampa** [reply]
Thank you very much. I got to the same conclusions.
